NextcloudTalk/UserStatusMessageSwiftUIView.swift

```diff
@@ -0,0 +1,369 @@
+            Button("Don't clear") {
+                selectedClearAtString = "Don't clear"
+                showClearAtAlert = false
+                setClearAt(clearAt: NSLocalizedString("Don't clear", comment: ""))
+            }
+            Button("30 minutes") {
+                selectedClearAtString = "30 minutes"
+                showClearAtAlert = false
+                setClearAt(clearAt: NSLocalizedString("30 minutes", comment: ""))
+            }
+            Button("1 hour") {
+                selectedClearAtString = "1 hour"
+                showClearAtAlert = false
+                setClearAt(clearAt: NSLocalizedString("1 hour", comment: ""))
+            }
+            Button("4 hours") {
+                selectedClearAtString = "4 hours"
+                showClearAtAlert = false
+                setClearAt(clearAt: NSLocalizedString("4 hours", comment: ""))
+            }
+            Button("Today") {
+                selectedClearAtString = "Today"
+                showClearAtAlert = false
+                setClearAt(clearAt: NSLocalizedString("Today", comment: ""))
+            }
+            Button("This week") {
+                selectedClearAtString = "This week"
+                showClearAtAlert = false
+                setClearAt(clearAt: NSLocalizedString("This week", comment: ""))
+            }
+        }
+        .alert(isPresented: $showErrorAlert) {
+            Alert(title: Text(errorAlertTitle), message: Text(errorAlertMessage), dismissButton: .cancel())
+        }
+    }
+
+    func getStatus() {
+        isLoading = true
+        NCAPIController.sharedInstance().setupNCCommunication(for: NCDatabaseManager.sharedInstance().activeAccount())
+        NextcloudKit.shared.getUserStatus { _, clearAt, icon, message, _, _, _, _, _, _, error in
+            if error.errorCode == 0 {
+                selectedIcon = icon ?? "😀"
+                selectedMessage = message ?? ""
+                selectedClearAt = clearAt?.timeIntervalSince1970 ?? 0
+                selectedClearAtString = getPredefinedClearStatusText(clearAt: clearAt, clearAtTime: nil, clearAtType: nil)
+            }
+        }
+        NextcloudKit.shared.getUserStatusPredefinedStatuses { _, userStatuses, _, error in
+            if error.errorCode == 0 {
+                statusPredefinedStatuses = userStatuses!
+                withAnimation {
+                    isLoading = false
+                }
+            }
+        }
+    }
+
+    func setActiveUserStatus() {
+        if !customStatusSelected {
+            NextcloudKit.shared.setCustomMessagePredefined(messageId: selectedPredifinedStatus!.id!, clearAt: selectedClearAt) { _, error in
+                if error.errorCode == 0 {
+                    //   let clearAtDate = NSDate(timeIntervalSince1970: selectedClearAt)
+                    //   self.delegate?.didSetStatusMessage(icon: self.predefinedStatusSelected?.icon, message: self.predefinedStatusSelected?.message, clearAt: clearAtDate)
```

Comment:
Do we need to handle this correctly? Currently we just dismiss, but it was updated in the old version?

Reply:
The UI is updated automatically and setting to server works just by calling.